annubes/task.py

```diff
@@ -265,6 +291,65 @@ def plot_trials(self, n_plots: int = 1) -> go.Figure:
         fig.update_layout(height=1300, width=900, title_text="Trials")
         return fig
 
+    def _check_str(self, name: str, value: Any) -> None:  # noqa: ANN401
+        if not isinstance(value, str):
+            msg = f"`{name}` must be a string"
+            raise TypeError(msg)
```

Comment:
As it is, this doesn't work in the following cases:
- `stim_intensities`: `_check_positive_float` is currently used on each element of `stim_intensities`, not on the whole list.
- `iti`: Similarly, when `iti` is a tuple, the `_check_int_positive` is done on each element of the tuple.
- `ntrials`" it is not an attribute, so it can't be found by `vars(self)[attr]`.
I think in general make this more modularised is not so trivial, so if we want to do it is better to create a new issue and PR.